src/maliput_viz/flags/gflags.cc

```diff
@@ -0,0 +1,109 @@
+
+#include "maliput_viz/flags/gflags.h"
+
+#include <iostream>
+#include <sstream>
+#include <string>
+
+#include <maliput/common/filesystem.h>
+
+// Gflags
+DEFINE_int32(verbosity, 3, "Verbosity is set within a range between [0-4]");
+DEFINE_string(yaml_file_path, "", "Path to the configuration YAML file");
+
+// Validators
+static bool ValidateVerbosity(const char* flagname, int32_t value) { return value >= 0 && value <= 4; }
+static bool ValidateConfigFilePath(const char* flagname, const std::string& value) {
+  if (!value.empty()) {
+    maliput::common::Path path{value};
+    if (!path.exists() || !path.is_file()) {
+      std::cout << "Please verify that '" << value << "' is a valid file path." << std::endl;
+      return false;
+    }
+  }
+  return true;
+}
+
+DEFINE_validator(verbosity, &ValidateVerbosity);
+DEFINE_validator(yaml_file_path, &ValidateConfigFilePath);
+
+namespace maliput {
+namespace viz {
+namespace flags {
+namespace {
+
+// Returns a string with the usage message.
+std::string GetUsageMessage() {
+  std::stringstream ss;
+  ss << "Application for maliput road network visualization" << std::endl << std::endl;
+  ss << "  maliput_viz [--<flag_1>] [--<flag_2>] .. [--<flag_n>] " << std::endl << std::endl;
+  ss << " * Summary: " << std::endl;
+  ss << "    This application offers a visualizer for the maliput road networks " << std::endl;
+  ss << "    It provides a GUI for interactively select a maliput backend and load the road network" << std::endl;
+  ss << "    passing the correspondent parameters. " << std::endl;
+  ss << "    In addition, a YAML configuration file could be passed via command line for setting up the visualizer "
+        "beforehand."
+     << std::endl
+     << std::endl;
+  ss << " * Examples of use: " << std::endl;
+  ss << "    1. Running the visualizer: " << std::endl;
+  ss << "       $ maliput_viz" << std::endl;
+  ss << "    2. Running the visualizer with a higher verbosity level: " << std::endl;
+  ss << "       $ maliput_viz --verbosity=4" << std::endl;
+  ss << "    3. Running the visualizer with a configuration file: " << std::endl;
+  ss << "       $ maliput_viz --yaml_file_path=<path-to-yaml-file> " << std::endl << std::endl;
+  ss << " * YAML file configuration: " << std::endl;
+  ss << "    The --yaml_file_path flag expects a valid path to a YAML file." << std::endl;
+  ss << "    The YAML file should contain the configuration for loading the road network." << std::endl;
+  ss << "    The structure of the YAML file should be: " << std::endl;
+  ss << "     |  maliput_viz:" << std::endl;
+  ss << "     |    maliput_backend: <backend_name> " << std::endl;
+  ss << "     |    parameters: " << std::endl;
+  ss << "     |       <key_1>: <value_1> " << std::endl;
+  ss << "     |       <key_2>: <value_2> " << std::endl;
+  ss << "     |       ...  " << std::endl;
+  ss << "     |       <key_N>: <value_N> " << std::endl << std::endl;
```

Comment:
Why isn't it a string literal?

https://en.cppreference.com/w/cpp/language/string_literal

Reply:
Yeah, right.
Done.